src/txrequest.cpp

```diff
@@ -678,6 +678,18 @@ class TxRequestTracker::Impl {
         if (it != m_index.get<ByPeer>().end()) MakeCompleted(m_index.project<ByTxHash>(it));
     }
 
+    bool ExpectedTx(NodeId peer, const uint256& txhash)
+    {
+        auto it_hash = m_index.get<ByTxHash>().lower_bound(ByTxHashView{txhash, State::REQUESTED, 0});
+        // We need to traverse all the REQUESTED/COMPLETED announcements to verify we effectively
+        // requested the txhash from this peer.
+        while (it_hash != m_index.get<ByTxHash>().end() && it_hash->m_txhash == txhash) {
+            if (it_hash->m_peer == peer) return true;
+            ++it_hash;
+        }
```

Comment:
I'm just trying to understand the boost multi-index data structure, and want to do due diligence whenever I see a `while` loop: This will iterate only as many times as we have actively requested a given TX, correct? So the most number of iterations possible is the total number of peers we have? Because the TX hash is specified, so it's not like we run through the entire data structure.